### ISSUE TITLE
problem with subtitle file name?

### ISSUE THREAD

**Issue by nvoynov** [eregon/suby#13]
Hello, eregon

I have just tried suby for "The 100" TV Show.

Command `$ suby The.100.s03e10` returned me file by name The.100.srt. And I can't find season and episode.

Command `$ suby The.100.s03e10.WEBDLRip` returned me file named in right way The.100.s03e10.srt

I think the case without season and episode isn't correct. Or it always required media file extension?

**Comment by nvoynov** [comment]
Also I have video files with two languages and they are named with suffix by translation studio, like quantico.s01e15.WEBDLRip.NewStudio.TV.avi. When I just copy file name to console command I have next:
```
suby Quantico.s01e14.WEBDLRip.NewStudio.TV.avi
Quantico.s01e14.WEBDLRip.NewStudio.TV.avi
  OpenSubtitles
The download of the subtitles failed for Quantico.s01e14.WEBDLRip.NewStudio.TV.avi:
NoMethodError: undefined method `[]' for nil:NilClass
C:/ruby/Ruby22/lib/ruby/gems/2.2.0/gems/suby-0.4.4/lib/suby/downloader/opensubtitles.rb:37:in `download_url'
C:/ruby/Ruby22/lib/ruby/gems/2.2.0/gems/suby-0.4.4/lib/suby/downloader.rb:83:in `download'
C:/ruby/Ruby22/lib/ruby/gems/2.2.0/gems/suby-0.4.4/lib/suby.rb:59:in `try_downloader'
C:/ruby/Ruby22/lib/ruby/gems/2.2.0/gems/suby-0.4.4/lib/suby.rb:45:in `block in download_subtitles_for_file'
C:/ruby/Ruby22/lib/ruby/gems/2.2.0/gems/suby-0.4.4/lib/suby.rb:44:in `each'
C:/ruby/Ruby22/lib/ruby/gems/2.2.0/gems/suby-0.4.4/lib/suby.rb:44:in `find'
C:/ruby/Ruby22/lib/ruby/gems/2.2.0/gems/suby-0.4.4/lib/suby.rb:44:in `download_subtitles_for_file'
C:/ruby/Ruby22/lib/ruby/gems/2.2.0/gems/suby-0.4.4/lib/suby.rb:30:in `block in download_subtitles'
C:/ruby/Ruby22/lib/ruby/gems/2.2.0/gems/suby-0.4.4/lib/suby.rb:21:in `each'
C:/ruby/Ruby22/lib/ruby/gems/2.2.0/gems/suby-0.4.4/lib/suby.rb:21:in `download_subtitles'
C:/ruby/Ruby22/lib/ruby/gems/2.2.0/gems/suby-0.4.4/bin/suby:30:in `<top (required)>'
```

but if I edit file name and remove suffix - all things are OK - I have right subtitle file.

**Comment by nvoynov** [comment]
The next interesting thing could be redirect subtitle file to stream, something like for shift time or so on. Suppose, I have standard movie files with advert from translation studio, and this advert usually have fixed time.

I'm not sure for syntax, but it might be something like `subshift 15 < suby my.movi.avi`

**Comment by eregon** [comment]
* Yes, currently the extension is required (the file does not need to exist for some downloaders, but it's always better if available).
* Could you try the newer version? (4.5)
* Do you mean printing the filename of the sub/srt file? Would it work with multiple movie files?

**Comment by pgaspar** [comment]
@eregon I'm also seeing the error @nvoynov describes.

Could you please publish version `0.4.5` to rubygems? It's not available there: https://rubygems.org/gems/suby.

As for the sub shift suggestion - that's a cool idea! There are some times where you get a subtitle that is shifted (like 5 seconds late). It would be great to have the option to correct that shift, but it may very well be out of scope for suby. I like the idea either way!

Actually, I just ran a search on github and I could find a few already 😃 https://github.com/search?utf8=%E2%9C%93&q=subtitle+shifter

**Comment by eregon** [comment]
@pgaspar You are right, I forgot to push the gem somehow.
I also released 0.4.6 with the latest commits, hopefully this works for you :smiley: 

Re shifting subtitles, it would be interesting but what would be truly awesome is to automatically match the text with the audio stream of the video file. In my experience, it's rarely just a constant shift. How to do that in practice is an open question though.
